### PR TITLE
TrustCommerce: Update `authorization_from` to handle `store` response

### PR DESCRIPTION
**Pull request by jherreraa**
## Summary:

Fix Store/Unstore features

Remote tests
-------------------------------------------------------------------------- 
22 tests, 80 assertions, 3 failures, 0 errors, 0 pendings, 
0 omissions, 0 notifications 
86.3636% passed

Failing tests not related with changes

unit tests
-------------------------------------------------------------------------- 
20 tests, 64 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/trust_commerce.rb [hunk 1]
```diff
@@ -476,7 +476,7 @@ def message_from(data)
       end
 
       def authorization_from(action, data)
-        authorization = data['transid']
+        action == 'store' ? authorization = data['billingid'] : authorization = data['transid']
```
Comment (sinourain) [inline lib/active_merchant/billing/gateways/trust_commerce.rb:479]: authorization can be defined only once:
`authorization = action == 'store' ? data['billingid'] : data['transid']`

**Review by sinourain — DISMISSED** [review]
LGTM just a minor change

File: test/remote/gateways/remote_trust_commerce_test.rb [hunk 2]
```diff
@@ -182,6 +182,13 @@ def test_failed_store
     assert_bad_data_response(response)
   end
 
+  def test_successful_unstore
+    assert store = @gateway.store(@credit_card)
+    assert_equal 'approved', store.params['status']
+    assert response = @gateway.unstore(store.params['billingid'])
+    assert_success response
+  end
+
```
Comment (Heavyblade) [inline test/remote/gateways/remote_trust_commerce_test.rb:191]: @jherreraa just to have a better remote tests and to ensure that store works as expected, please add a test that can use an stored payment method to do a purchase/authorize

File: lib/active_merchant/billing/gateways/trust_commerce.rb (hunk 1 above)
Comment (Heavyblade) [reply lib/active_merchant/billing/gateways/trust_commerce.rb:479]: maybe is a good time for a refactor ?, something like:

```ruby
def authorization_from(action, data)
  case action
  when 'store'
    data['billingid']
  when *VOIDABLE_ACTIONS
    "#{data['transid']}|#{action}"
  else
    data['transid']
  end
end
```

**Review by Heavyblade — CHANGES_REQUESTED** [review]
@jherreraa we are close, leave you a couple of comments.

File: test/remote/gateways/remote_trust_commerce_test.rb (hunk 2 above)
Comment (jherreraa) [reply test/remote/gateways/remote_trust_commerce_test.rb:191]: Test added! Please check it again!

File: lib/active_merchant/billing/gateways/trust_commerce.rb (hunk 1 above)
Comment (jherreraa) [reply lib/active_merchant/billing/gateways/trust_commerce.rb]: Suggestion taken!

**Review by jherreraa — COMMENTED** [review]
solved! Thanks for the feedback

**Review by naashton — APPROVED** [review]
This change looks good! 👍

**Pull request merged**